common/error.h

```diff
@@ -133,7 +133,14 @@ class ErrorBuilder {
 
   // Accumulates string message.
   template <typename T>
-  [[nodiscard]] auto operator<<(const T& message) -> ErrorBuilder& {
+  [[nodiscard]] auto operator<<(const T& message) && -> ErrorBuilder&& {
+    *out_ << message;
+    return std::move(*this);
+  }
+
+  // Accumulates string message for an lvalue error builder.
+  template <typename T>
+  auto operator<<(const T& message) & -> ErrorBuilder& {
```

Comment:
```suggestion
  [[nodiscard]] auto operator<<(const T& message) & -> ErrorBuilder& {
```

(for consistency)

Reply:
That would defeat the purpose of this change, which is to allow `builder << "something";` rather than `static_cast<void>(builder << "something");` while still disallowing `ErrorBuilder(loc) << "something";`. I've extended the comments to try to make this a little clearer.